pkg/volume/projected/projected.go

```diff
@@ -305,6 +305,11 @@ func (s *projectedVolumeMounter) collectData() (map[string]volumeutil.FileProjec
 				errlist = append(errlist, fmt.Errorf("pod request ServiceAccountToken projection but the TokenRequestProjection feature was not enabled"))
 				continue
 			}
+
+			if len(s.pod.Spec.ServiceAccountName) == 0 {
+				errlist = append(errlist, fmt.Errorf("pod request ServiceAccountToken projection but the serviceaccount name is empty"))
```

Comment:
hi @WanLinghao, it would be better to move this into pod/podtemplate validation. WDYT? leave only runtime errors here.